### PR TITLE
feat: updated NINO to use regex in api config

### PR DESCRIPTION
**Pull request by merlinc**
### What changed

Allow any NINO to be used in the UI, except for the ones in use by the various failure paths

### Issue tracking

- [OJ-2448](https://govukverify.atlassian.net/browse/OJ-2448)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2448]: https://govukverify.atlassian.net/browse/OJ-2448?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kiejgordon** [comment]
Had to explicitly state the retry nino value should not be a success

File: tests/imposter/private-api-config.yaml [hunk 1]
```diff
@@ -113,7 +117,7 @@ resources:
     requestBody:
       allOf:
         - jsonPath: $.nino
-          value: "AA123455D"
+          value: "EE123456A"
```
Comment (Caitlin-cooling) [inline tests/imposter/private-api-config.yaml:120]: If this is a scenario that we need do any tests need to be updated to use it?

Comment (MarieseMikely) [reply tests/imposter/private-api-config.yaml:120]: I think this change might have been a mistake - will update

Comment (kiejgordon) [reply tests/imposter/private-api-config.yaml:120]: Was changing values when testing and forgot to revert, didn't affect the tests but have changed back to original, Thanks!

**Review by Caitlin-cooling — DISMISSED** [review]
Looks good to me - thanks for the changes!

**Review by liamStockwellGDS — APPROVED** [review]

**Pull request merged**